File: domain/src/main/java/com/icehousecorp/maunorafiq/domain/weather/repository/WeatherRepository.java
package com.icehousecorp.maunorafiq.domain.weather.repository;

import com.icehousecorp.maunorafiq.domain.weather.Weather;

import rx.Observable;

/**
 * Created by maunorafiq on 12/2/16.
 */

public interface WeatherRepository {
    Observable<Weather> currentWeather(String city);

    Observable<Weather> currentWeather(Double lat, Double lon);
}

File: domain/src/main/java/com/icehousecorp/maunorafiq/domain/weather/repository/WeatherRepository.kt
package com.icehousecorp.maunorafiq.domain.weather.repository

import com.icehousecorp.maunorafiq.domain.weather.Weather

import rx.Observable

/**
 * Created by maunorafiq on 12/2/16.
 */

interface WeatherRepository {
    fun currentWeather(city: String): Observable<Weather>

}
